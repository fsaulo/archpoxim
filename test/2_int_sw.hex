0xDC00000A
0xDC000006
0xDC000005
0xDC000004
0x00000000
0x00000000
0x00000000
0x00000000
0x083A0000
0x085B0000
0x80000000
0x03C07FFC
0xFC000005
0x87E10001
0x10220400
0xF0F0F0F0
0xFC000000