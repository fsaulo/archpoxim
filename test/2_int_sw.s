// Segmento de codigo
.text
    // Tabela de vetor de interrupcao
    init:
        bun main
        bun isr
        bun isr
        bun isr
        .align 5
    // Rotina de tratamento de interrupcao
    isr:
        // R1 = CR
        mov r1, cr
        // R2 = IPC
        mov r2, ipc
        // Retorno de ISR
        reti
    // Funcao principal
    main:
        // SP = 32 KiB
        mov sp, 0x7FFC
        // Interrupcao de software 5
        int 5
        // Habilitando interrupcao (IE = 1)
        sbr sr[1]
        // Divisao por zero
        div r1, r2, r0
        // Instrucao invalida
        .4byte 0xF0F0F0F0
        // Finalizacao de execucao
        int 0