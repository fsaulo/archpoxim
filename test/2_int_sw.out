[START OF SIMULATION]
0x00000000:	bun 10                   	PC=0x0000002C
0x0000002C:	mov sp,32764             	SP=0x00007FFC
0x00000030:	int 5                    	CR=0x00000005,PC=0x0000000C
[SOFTWARE INTERRUPTION]
0x0000000C:	bun 4                    	PC=0x00000020
0x00000020:	add r1,cr,r0             	R1=CR+R0=0x00000005,SR=0x00000000
0x00000024:	add r2,ipc,r0            	R2=IPC+R0=0x00000030,SR=0x00000000
0x00000028:	reti                     	IPC=MEM[0x00007FF4]=0x00000000,CR=MEM[0x00007FF8]=0x00000000,PC=MEM[0x00007FFC]=0x00000034
0x00000034:	sbr sr[1]                	SR=0x00000002
0x00000038:	div r0,r1,r2,r0          	R0=R2%R0=0x00000000,R1=R2/R0=0x00000005,SR=0x00000022
[SOFTWARE INTERRUPTION]
0x00000008:	bun 5                    	PC=0x00000020
0x00000020:	add r1,cr,r0             	R1=CR+R0=0x00000000,SR=0x00000062
0x00000024:	add r2,ipc,r0            	R2=IPC+R0=0x00000038,SR=0x00000022
0x00000028:	reti                     	IPC=MEM[0x00007FF4]=0x00000000,CR=MEM[0x00007FF8]=0x00000000,PC=MEM[0x00007FFC]=0x0000003C
[INVALID INSTRUCTION @ 0x0000003C]
[SOFTWARE INTERRUPTION]
0x00000004:	bun 6                    	PC=0x00000020
0x00000020:	add r1,cr,r0             	R1=CR+R0=0x0000003C,SR=0x00000026
0x00000024:	add r2,ipc,r0            	R2=IPC+R0=0x0000003C,SR=0x00000026
0x00000028:	reti                     	IPC=MEM[0x00007FF4]=0x00000000,CR=MEM[0x00007FF8]=0x00000000,PC=MEM[0x00007FFC]=0x00000040
0x00000040:	int 0                    	CR=0x00000000,PC=0x00000000
[END OF SIMULATION]
